build/package/scripts/build-java.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+set -eu
+
+OUTPUT_DIR="docker"
+WORKING_DIR="."
+ROOT_DIR=$(pwd)
+ARTIFACTS_DIR=$ROOT_DIR/.ods/artifacts
+ARTIFACT_PREFIX=
+DEBUG="false"
+GRADLE_ADDITIONAL_TASKS=
+GRADLE_OPTIONS=
+
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+
+    --working-dir) WORKING_DIR="$2"; shift;;
+    --working-dir=*) WORKING_DIR="${1#*=}";;
+
+    --output-dir) OUTPUT_DIR="$2"; shift;;
+    --output-dir=*) OUTPUT_DIR="${1#*=}";;
+
+    --gradle-additional-tasks) GRADLE_ADDITIONAL_TASKS="$2"; shift;;
+    --gradle-additional-tasks=*) GRADLE_ADDITIONAL_TASKS="${1#*=}";;
+
+    # Gradle project properties ref: https://docs.gradle.org/current/userguide/build_environment.html#sec:gradle_configuration_properties
+    # Gradle options ref: https://docs.gradle.org/current/userguide/command_line_interface.html
+    --gradle-options) GRADLE_OPTIONS="$2"; shift;;
+    --gradle-options=*) GRADLE_OPTIONS="${1#*=}";;
+
+  *) echo "Unknown parameter passed: $1"; exit 1;;
+esac; shift; done
+
+if [ -z "${WORKING_DIR}" ]; then
+  WORKING_DIR="${ROOT_DIR}"
+fi
+
+if [ "${DEBUG}" == "true" ]; then
+  set -x
+  echo
+  echo "Debug mode enabled (DEBUG=true)"
+  echo "Generating debug logs ..."
+  echo "... current folder:"
+  pwd
+  echo "... folder content:"
+  ls -lart
+  echo
+fi
+
+cd "${WORKING_DIR}"
+echo "Working on Java/Gradle project in ${WORKING_DIR} ..."
+echo
+echo "... gladew version: "
+./gradlew -version
+
+echo
+echo "Note on Nexus: following env vars NEXUS_HOST, NEXUS_USER and NEXUS_PASSWORD are available and could be read by your gradle script"
+echo
+echo "Building (Compile and Test) ..."
+# TODO decide if param odsPipelineOutputDir should be used or renamed!
```

Comment:
Let's discuss this. I guess the issue is that the Gradle file hardcodes where to place the output, right?

Reply:
yes, the default location is `project.distsDir` (https://docs.gradle.org/current/dsl/org.gradle.api.tasks.bundling.Jar.html#org.gradle.api.tasks.bundling.Jar:destinationDirectory)

Reply:
Couldn't we set `project.distsDir` dynamically? If so, are there reasons not to do it?

Reply:
Currently we are graving in the script `build-java.sh` the unit test results and coverage results from a specific folders. 
If we follow same approach, this would result in that all produced jars from `build/libs` will need to be copied to `OUTPUT_DIR`. What do you think?

Reply:
I pushed a different approach. The script `build-java.sh` export a env var called `ODS_OUTPUT_DIR`. The scrips echo a warning now and explains the availability of this env var. The project gradle script could read this env var to copy the jar file to it.